Fix HTTP status codes

annielinnik opened the pull request:
- Renamed custom HTTP status phrases to follow proper conventions (e.g., 403 Forbidden)
- Added detailed error messages in response bodies where needed (e.g., {error: "userId is missing"})
- Added decorator for GitcoinClaimer

liringlas commented on src/modules/gitcoin-claimer/GitcoinClaimer.ts:
```diff
@@ -426,14 +397,15 @@ export class GitcoinClaimer {
+    const address = getValidatedAddressSchema(body);
     // Check address score
-    const { value: score } = await this.getAddressScore(body);
+    const { value: score } = await this._gitcoinApi.getScore(address);
```
May just use `this.getAddressScore`, it gets `address` from the `body` the same way.

liringlas replied:
Maybe also make `this.getAddressScore` return `address` if you need it where you call it.

annielinnik replied:
`getAddressScore` validates body and retrieves address from it, and then we again validate it and retrieve address. Don't see the point of running it twice.

annielinnik committed:
fix: standardize HTTP status codes and add detailed error messages
- Renamed custom HTTP status phrases to follow proper conventions (e.g., 403 Forbidden)
- Added detailed error messages in response bodies where needed (e.g., {error: "userId is missing"})
- Removed excess log
